src/Makefile.am

```diff
@@ -808,8 +808,7 @@ tests_testMem_LDFLAGS = $(LIBADD_DL)
 
 check_PROGRAMS += tests/testCharacterSet
 tests_testCharacterSet_SOURCES = \
```

Comment:
I would very much like to see this PR removing _all_ headers that match the (finalized) removal logic. Are you still working on removing more files? If you are, thank you, and I will be out of your hair until you are done. If you are done, then I would like to understand why most of the remaining headers were not removed. If you need assistance with removing move files, please let me know.

FWIW, here are the currently remaining headers that I know about. Using the same removal logic, many of these should be removed. I have not checked whether any of these are reused.

```
compat/testPreCompiler.h
lib/tests/testRFC1738.h
src/tests/testACLMaxUserIP.h
src/tests/testAuth.h
src/tests/testBoilerplate.h
src/tests/testIpAddress.h
src/tests/testLookupTable.h
src/tests/testMem.h
src/tests/testRock.h
src/tests/testSBuf.h
src/tests/testSBufList.h
src/tests/testStatHist.h
src/tests/testStore.h
src/tests/testStoreController.h
src/tests/testStoreHashIndex.h
src/tests/testString.h
src/tests/testTokenizer.h
src/tests/testURL.h
src/tests/testUfs.h
src/tests/testUriScheme.h
src/tests/testYesNoNone.h
```

Reply:
> I would very much like to see this PR removing _all_ headers that match the (finalized) removal logic

That's my plan that's why the PR is in draft.
I'm currently prioritising fixing the portability and build issues I uncovered first, as these block other development. Once that is done, I'll carry this forward

Reply:
Great plan, thank you! No rush.

BTW, when posting a Draft PR, consider adding a GitHub comment to explain why the PR was marked as a draft -- disclosing your reasons would reduce the number of questions others may have to ask and you may have to answer...

Reply:
done